tests/phpunit/tests/speculative-loading/wpGetSpeculationRulesConfiguration.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * Tests for the wp_get_speculation_rules_configuration() function.
+ *
+ * @package WordPress
+ * @subpackage Speculative Loading
+ */
+
+/**
+ * @group speculative-loading
+ * @covers ::wp_get_speculation_rules_configuration
+ */
+class Tests_Speculative_Loading_wpGetSpeculationRulesConfiguration extends WP_UnitTestCase {
+
+	public function set_up() {
+		parent::set_up();
+
+		update_option( 'permalink_structure', '/%year%/%monthnum%/%day%/%postname%/' );
```

Comment:
Note there's a `set_permalink_structure()` helper method

Reply:
Thanks for pointing that out, didn't know. Looking at the method though, it does more than we need to for this - we don't actually care about flushing the rewrite rules as that doesn't matter for this functionality. We only need to update the option itself, so I think this works better as is.